mfs/inode.go

```diff
@@ -0,0 +1,54 @@
+package mfs
+
+import (
+	ipld "gx/ipfs/QmWi2BYBL5gJ3CiAiQchg6rn1A8iBsrWy51EYxvHVjFvLb/go-ipld-format"
+)
+
+// Inode abstracts the common characteristics of the MFS `File`
+// and `Directory`. All of its attributes are initialized at
+// creation and can't be modified.
+type Inode interface {
+
+	// Name of this `Inode` in the MFS path (the same value
+	// is also stored as the name of the DAG link).
+	Name() string
+
+	// Parent directory of this `Inode` (which may be the `Root`).
+	Parent() childCloser
+
+	// DagService used to store modifications made to the contents
+	// of the file or directory the `Inode` belongs to.
+	DagService() ipld.DAGService
+}
+
+// inode implements the `Inode` interface.
+type inode struct {
```

Comment:
Is this the appropriate name for the structure that implements the `Inode` interface?

Reply:
Yeah, thats what I would do.